### DISCUSSION TITLE
Problems ignoring go.mod in root

### DISCUSSION THREAD

**Discussion by dgrenner** [renovatebot/renovate#9875] [Request Help] [answered]
My project has a file structure like:
```
.
├── foo
│   └── go.mod
├── bar
│   └── go.mod
├── baz
│   └── go.mod
└── go.mod
```
I would like to ignore scan of `go.mod` in `root`, and the one in `foo` - but I want to scan the ones in `bar`  and `baz`.
When using:
```
{
  "extends": [
    "config:base"
  ],
  "ignorePaths": [
    "go.mod",
    "foo/go.mod"
  ]
}
```
the first `go.mod` will match all path variants, so nothing is scanned. I've also tried with `^go.mod`, but that didn't work either. Is there any `ignorePaths` or other setting that would cover this case?

**Comment by rarkins** [comment, marked as answer]
Try:

```
{
  "extends": [
    "config:base"
  ],
  "ignorePaths": [
    "go.mod*",
    "foo/go.mod"
  ]
}
```

**Comment by viceice** [comment]
I think better use:

```json
{
  "extends": [
    "config:base"
  ],
  "ignorePaths": [
    "+(go.mod)",
    "+(foo/go.mod)"
  ]
}
```

`ignorePaths` uses glob like [`matchPaths`](https://docs.renovatebot.com/configuration-options/#matchpaths) `+(...)` forces glob matching only.

https://github.com/renovatebot/renovate/blob/74d7691befa635c1220d8569936273bfb950e748/lib/workers/repository/extract/file-match.ts#L28-L35

**Reply by dgrenner** [reply]
I tried this, and works for `+(go.mod)`. I found that more clear than the other suggestion - with the `*` at the end it looks like it could match more things (even though that wouldn't cause an issue here). 
However with the folder separator included in `+(foo/go.mod)`, it does not match with minimatch. In the end I decided to handle that path without glob as suggested in the response from @rarkins, since I don't have any duplicated path in that case.

**Reply by viceice** [reply]
@rarkins  maybe add `ignoreFiles` option for exact matching? like we have for `matchFiles` ?
